core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -110,8 +110,23 @@ class KafkaRequestHandler(id: Int,
 
         case callback: RequestChannel.CallbackRequest =>
           try {
-            callback.originalRequest.callbackRequestDequeTimeNanos = Some(time.nanoseconds())
+            val originalRequest = callback.originalRequest
+            
+            // If we've already executed a callback for this request, reset the times and subtract the callback time from the 
+            // new dequeue time. This will allow calculation of multiple callback times.
+            // Otherwise, set dequeue time to now.
+            if (originalRequest.callbackRequestDequeueTimeNanos.isDefined) {
```

Comment:
Not sure if the 'if' clause is required, prevCallbacksTimeNanos would be 0 if there was no prev callback.

Reply:
I just wanted to avoid doing all this extra code. In my head, it made more sense to someone reading it.